staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/validator.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+
+	celtypes "github.com/google/cel-go/common/types"
+	v1 "k8s.io/api/admissionregistration/v1"
+)
+
+// validator implements the Validator interface
+type validator struct {
```

Comment:
Could a validator wrap an instance of `evaluator{}`? That would simplify the calling model slightly:

* NewValidator(Expressions) builds the compiler under-the-hood
* Validate() can be called directly against the inbound resource, meaning the caller does not need to be aware of the structure of how CEL is called/organized.

Reply:
wrapped `filter`